### DISCUSSION TITLE
G7 mismatch

### DISCUSSION THREAD

**Discussion by Pentagit** [NightscoutFoundation/xDrip#3674] [Bug reports] [answered]
Hello I can't connect new G7 Sensor to xdrip. Get mismatch - wait and scanning Errors. 

- Sensor code is correct, of course. 
- Xdrip version is current as of 19.09.2024
- Phone: Google Pixel 8 pro, latest Android
- qr code Navid scanned
- dexcom app not installed 
- using xdrip since 2017 with dexcom sensors (G5, G6, now G7)

I had the same problem a few weeks back already and solved it - by installing original dexcom app. Which really can't be the solution.
Now today I inserted a new G7 and got that mismatch again!
I unistalled xdrip, reinstalled (without importing my settings), scanned qr and it does not work.

This has not occured before and is a serious problem. My guess is that this is somehow related to android bluetooth stuff. 
I will now install diakem dexcom app to get data to androidaps and enable smb.

Please have a look into this problem thanks.

**Comment by Pentagit** [comment, marked as answer]
Nevermind. Sensor couldnt be detected by Dexcom app as well. BLE Scanner also didnt see it, propably defective sensor.

Still: why did xdrip give a mismatch?

Edit: New Sensor immediately connected and works.

**Reply by Waxwax462** [reply]
had exactly the same situation this morning at 05:00 AM. I will try now an third sensor.
But it would be extremly useful to get a "real error message "when the sensor is defective and not this misleading "mismatch - wait" error.

**Reply by Navid200** [reply]
You should never conclude that a sensor is defective just because xDrip fails.  You should always try Dexcom app as a verification step to confirm that.

xDrip has no way of verifying if a sensor is defective.  Neither does Dexcom app.  Both can  do the best they can to establish connectivity with a sensor.  That connectivity is a two way (handshake) process.  If you have already established connectivity to a sensor with another app on the same or another phone, there is no way you can establish connectivity to it with a second app because G7 will not be looking for connectivity any longer.  But, that does not mean there is anything wrong with the sensor.

In other words, if xDrip cannot get a handshake from a sensor, there could be different reasons, one of which is sensor malfunction.  But, there is no way to confirm that.
So, we cannot change xDrip to show sensor defect.
